src/app/entity/schema/entity-schema.service.spec.ts

```diff
@@ -0,0 +1,116 @@
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     ndb-core is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Entity } from '../entity';
+import {async} from '@angular/core/testing';
+import {EntitySchemaService} from './entity-schema.service';
+import {DatabaseField} from '../database-field.decorator';
+
+describe('EntitySchemaService', () => {
+  let entitySchemaService: EntitySchemaService;
+
+  beforeEach(async(() => {
+    entitySchemaService = new EntitySchemaService();
+  }));
+
+
+  it('schema:string converts to strings', function () {
+    class TestEntity extends Entity {
+      @DatabaseField() aString: string;
+    }
+    const id = 'test1';
+    const entity = new TestEntity(id);
+
+    const data = {
+      _id: 'test2',
+      aString: 192
+    };
+    entitySchemaService.loadDataIntoEntity(entity, data);
+
+    expect(entity.aString).toBe('192');
+
+    const rawData = entitySchemaService.transformEntityToDatabaseFormat(entity);
+    expect(rawData.aString).toBe('192');
+  });
+
+  it('schema:number converts to numbers', function () {
+    class TestEntity extends Entity {
+      @DatabaseField() aNumber: number;
+      @DatabaseField() aFloat: number;
+    }
+    const id = 'test1';
+    const entity = new TestEntity(id);
+
+    const data = {
+      _id: 'test2',
+      aNumber: '192',
+      aFloat: '1.68',
+    };
+    entitySchemaService.loadDataIntoEntity(entity, data);
+
+    expect(entity.aNumber).toBe(192);
+    expect(entity.aFloat).toBe(1.68);
+
+    const rawData = entitySchemaService.transformEntityToDatabaseFormat(entity);
+    expect(rawData.aNumber).toBe(192);
+    expect(rawData.aFloat).toBe(1.68);
+  });
+
+  it('schema:date converts to Date object', function () {
+    class TestEntity extends Entity {
+      @DatabaseField() defaultDate: Date = new Date();
+      @DatabaseField() otherDate: Date;
+    }
+    const id = 'test1';
+    const entity = new TestEntity(id);
+
+    const data = {
+      _id: 'test2',
+      otherDate: '2018-01-01',
+    };
+    entitySchemaService.loadDataIntoEntity(entity, data);
+
+    expect(entity.defaultDate.toDateString()).toBe((new Date()).toDateString());
+
+    expect(entity.otherDate.getFullYear()).toBe(2018);
+    expect(entity.otherDate.getMonth()).toBe(0);
+    expect(entity.otherDate.getDate()).toBe(1);
+
+    const rawData = entitySchemaService.transformEntityToDatabaseFormat(entity);
+    expect(rawData.otherDate).toBe(data.otherDate);
+  });
+
+  it('schema:month converts to correctly between string and Date objects', function () {
```

Comment:
Maybe change the description of this test

Reply:
Do you have a suggestion? Is the `schema:month` confusing - or which part do you stumble over?

Reply:
No, I just don't understand the sentence grammatically. I think you only need to remove the 'to'.

Reply:
oh, yes. You are absolutely right - don't know how I missed that after your first comment